doc/man1/x509.pod

```diff
@@ -458,14 +454,14 @@ specified then the extensions should either be contained in the unnamed
 L<x509v3_config(5)> manual page for details of the
 extension section format.
 
-=item B<-force_pubkey key>
+=item B<-force_pubkey filename>
 
-When a certificate is created set its public key to B<key> instead of the
-key in the certificate or certificate request. This option is useful for
-creating certificates where the algorithm can't normally sign requests, for
-example DH.
+When a certificate is created set its public key to the key in B<filename>
+instead of the key contained in the input or given with the B<-signkey> option.
+This option is useful for creating self-issued certificates that are not
+self-signed, for instance when the key cannot be used for signing, such as DH.
```

Comment:
Don't know whether this warrants a mention, but the option does more than just allow you to create self-issued certificates.  When the input is a CSR, it looks to me like we can replace the public key in the CSR with any key of our choice, and issue certificates where the subject and issuer are not the same. Basically, construct a CSR with some random key, inject the desired public key (even if we don't have the private half to generate an ordinary CSR) and then sign the corresponding certificate. This can actually be useful in some cases. Should it be mentioned here?

Reply:
Yes, this is now possible using `-force_pubkey`, though not with `-signkey` (which leads to a self-issued cert because if makes the issuer same as the subject), but with the `-CA` (and potentially `-CAkey`) options.

As soon as this PR is fully reviewed/accepted I plan to submit, based on it, a second PR for the `x509` app adding `-subj` and `-new` options like with the `req` app. Then use cases like the one you described above will be more direct/convenient to implement without needing an intermediate CSR with a temporary dummy key:
`-force_pubkey` to set the public key, `-subj` to set the desired subject DN, `-CA` (maybe with `-CAkey`) or `-signkey` to set the private key used for signing and the issuer DN, and optionally `-extfile` to define any desired extensions. 
I'd say then is the optimal opportunity to mention this use case as an explicit example.

Reply:
@mattcaswell What do you think, leave the documentation as proposed for now, or make it be explicit about what becomes possible already with this PR?  If the follow-on PR is fast on the heels of this one, perhaps OK to defer, if this is just for "master", perhaps no rush?

Reply:
You can preview my planned follow-on PR here: https://github.com/openssl/openssl/compare/master...DDvO:add_new_and_subj_to_x509

Among others, this is the tentative addition to the doc of `-force_pubkey` I have in mind here:
> It can also be used in conjunction with b<-new> and B<-subj> to directly
generate a certificate containing any desired public key.

Should we also mention there that, depending on whether `-signkey` or `-CA(key)` is used, the new cert will be self-issued or have the subject of the cert given via `-CA` as its issuer?

Reply:
Happy to defer it for now if there is a follow on PR in the works.

Reply:
I've just happily submitted the mentioned follow-on PR #8193 :)